common/src/main/java/com/microsoft/identity/common/internal/broker/ipc/BrokerOperationBundle.java

```diff
@@ -63,11 +69,42 @@
     @Getter
     @Nullable final private Bundle bundle;
 
-//    public String getAccountManagerOperationKey(){
+    //    public String getAccountManagerOperationKey(){
 //        // TODO
 //    }
 //
-//    public String getContentProviderUriPath(){
-//        // TODO
-//    }
+    public String getContentProviderPath() throws BrokerCommunicationException {
+        final String methodName = ":getContentProviderUriPath";
+
+        switch (operation) {
+            case MSAL_HELLO:
+                return BrokerContentProvider.HELLO_PATH;
+
+            case MSAL_GET_INTENT_FOR_INTERACTIVE_REQUEST:
+                return BrokerContentProvider.ACQUIRE_TOKEN_INTERACTIVE_PATH;
+
+            case MSAL_ACQUIRE_TOKEN_SILENT:
+                return BrokerContentProvider.ACQUIRE_TOKEN_SILENT_PATH;
+
+            case MSAL_GET_ACCOUNTS:
+                return BrokerContentProvider.GET_ACCOUNTS_PATH;
+
+            case MSAL_REMOVE_ACCOUNT:
+                return BrokerContentProvider.REMOVE_ACCOUNTS_PATH;
+
+            case MSAL_GET_DEVICE_MODE:
+                return BrokerContentProvider.GET_DEVICE_MODE_PATH;
+
+            case MSAL_GET_CURRENT_ACCOUNT_IN_SHARED_DEVICE:
+                return BrokerContentProvider.GET_CURRENT_ACCOUNT_SHARED_DEVICE_PATH;
+
+            case MSAL_SIGN_OUT_FROM_SHARED_DEVICE:
+                return BrokerContentProvider.SIGN_OUT_FROM_SHARED_DEVICE_PATH;
```

Comment:
Could these paths be derived out of the operation enum? (so we don't need to have this switch)

Reply:
not really. the operation could be converted to a content provider path, or account manager key, or something else, depending on the communication mechanism. so it's not a 1:1 relationship.

Reply:
Yeah what I meant is that should we store each of those mappings inside the enum? (rather than using constants from a different file)

Reply:
So the operation contains mapping for each communication mechanism

Reply:
I'd say no, but I don't feel strong on this.

The reason is that the same constants are also used on the server (broker) side, and that side should not be aware of these operation object (or it should?)